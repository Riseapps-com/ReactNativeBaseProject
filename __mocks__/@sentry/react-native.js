module.exports = {
  ...jest.requireActual('@sentry/react-native'),

  init: jest.fn(),
  captureMessage: jest.fn(),
  captureException: jest.fn(),
};
