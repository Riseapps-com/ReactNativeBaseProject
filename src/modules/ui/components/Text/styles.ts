import { getFont } from '~assets/fonts';
import { sizes, styleSheetFactory } from '~theme';

export default styleSheetFactory(theme => ({
  text: {
    color: theme.text,
  },

  regular: {
    fontFamily: getFont('quicksand-regular'),
  },
  bold: {
    fontFamily: getFont('quicksand-bold'),
  },

  smallest: {
    fontSize: sizes.SMALLEST_FONT_SIZE,
  },
  smaller: {
    fontSize: sizes.SMALLER_FONT_SIZE,
  },
  normal: {
    fontSize: sizes.BASE_FONT_SIZE,
  },
  bigger: {
    fontSize: sizes.BASE_FONT_SIZE,
  },
  biggest: {
    fontSize: sizes.BASE_FONT_SIZE,
  },
}));
