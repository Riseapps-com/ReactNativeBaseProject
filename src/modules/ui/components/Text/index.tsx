import React from 'react';
import { Text as NativeText, TextStyle } from 'react-native';

import { TextFontStyle, TextSize } from '../../types';
import styles from './styles';

export type TextProps = React.ComponentProps<typeof NativeText> & {
  fontStyle?: TextFontStyle;
  size?: TextSize;
  style?: TextStyle;
};

const DEFAULT_FONT_STYLE = 'regular';
const DEFAULT_FONT_SIZE = 'normal';

/**
 * Renders a text with various styles. Text properties like size and weight can be passed via props.
 *
 * Should be used for the various titles used in the app to ensure consistency.
 */
const Text: React.FC<TextProps> = (props: TextProps) => {
  const { fontStyle, size, style, ...restProps } = props;

  const textStyles: TextStyle[] = [
    styles.text,
    styles[fontStyle || DEFAULT_FONT_STYLE],
    styles[size || DEFAULT_FONT_SIZE],
    style || {},
  ];

  return <NativeText style={textStyles} {...restProps} />;
};

export default React.memo(Text);
