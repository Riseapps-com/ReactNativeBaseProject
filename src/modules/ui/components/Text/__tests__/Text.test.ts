import { RenderAPI } from '@testing-library/react-native';

import { renderComponent } from '~modules/tests';

import Text, { TextProps } from '../index';

const children = 'Test text';

const defaultProps: TextProps = {
  children,
};

const renderText = (props: TextProps = {}) =>
  renderComponent(Text, { ...defaultProps, ...props }) as RenderAPI;

describe('ui', () => {
  describe('<Text />', () => {
    it('renders with correct text', () => {
      const text = renderText();

      expect(text.getByText(children)).toBeTruthy();
    });
  });
});
