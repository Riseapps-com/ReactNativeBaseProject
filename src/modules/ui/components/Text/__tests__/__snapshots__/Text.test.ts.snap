// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui <Text /> renders <Text /> 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      Object {
        "textAlign": "left",
      },
      Array [
        Object {
          "color": "#4A4E4D",
        },
        Object {
          "fontFamily": "SFProDisplay-Regular",
        },
        Object {
          "fontSize": 14,
        },
        undefined,
      ],
    ]
  }
>
  Hello world!
</Text>
`;
